http/http/src/main/java/io/helidon/http/PathMatchers.java

```diff
@@ -60,9 +61,22 @@ public static PathMatcher exact(String pathToMatch) {
      *
      * @param pathToMatch the path must be prefixed by the provided path
      * @return prefix match path matcher
+     * @deprecated use {@link #prefix(String, String)} instead to preserve original pattern
      */
+    @Deprecated(forRemoval = true, since = "4.3.0")
     public static PathMatcher prefix(String pathToMatch) {
-        return new PrefixPathMatcher(fixPrefix(pathToMatch));
+         return prefix(pathToMatch, null);
```

Comment:
`null` is not allowed in Helidon as a parameter.
Also this method should not be removed, it still makes sense (as you can create a custom prefix path matcher using).
The default behavior can be to add `/*` to it for the `pathPattern`.

Reply:
I wasn't sure adding "/*" was correct every time